src/_pytest/pytester.py

```diff
@@ -379,15 +380,29 @@ def LineMatcher_fixture(request: FixtureRequest) -> "Type[LineMatcher]":
 
 
 @pytest.fixture
-def testdir(request: FixtureRequest, tmp_path_factory) -> "Testdir":
+def test_path(request: FixtureRequest, tmp_path_factory) -> "TestPath":
```

Comment:
Dunno why but the name `test_path` feels a bit anemic to me. I do prefer `testpath` -- without the `_` it feels more like a proper noun, rather than a variable name. I'm thinking the name `pytester` itself is also pretty fitting for this. WDYT?

Reply:
I agree `test_path` doesn't sit very well, but I'm following our more modern approach of using `_` adequately in names, for example `tmp_path`. As much as I don't like `test_path` very much, I would rather prefer consistency... but not 100% sure either.

Reply:
Right, consistency with `temp_path` means it should be `test_path`, not `testpath` (for the same reason above I would have preferred if it were `tmppath` but that's irrelevant now).

The name `pytester` still works though, unless you don't like it of course.

Reply:
> The name pytester still works though, unless you don't like it of course.

Sure, but I'm not sure if you want me to do anything here though, as `pytester` is the name of the module and I didn't touch that.

Reply:
>  I'm thinking the name pytester itself is also pretty fitting for this. WDYT?

OH you are suggesting naming the fixture `pytester`, sorry missed that bit! 😆 

That's wild, but I definitely agree: the fixture is much more about running and checking tests than a "directory containing tests". 😁 

@RonnyPfannschmidt what do you think?

Reply:
Im not opposed to a name change, but I do wonder about turning it into a composition

Reply:
FWIW :+1: on naming the fixture `pytester`. I was always confused about the plugin being called `pytester` but the fixture being `testdir`, and I agree it's too common/vague as a name (it could be a directory containing test data or something). Seems like the right way forward, and hopefully we can then deprecate/remove `testdir` at some point in the (far) future?